src/main/kotlin/com/ajaxproject/warehouse/service/ProductServiceImpl.kt

```diff
@@ -1,73 +1,75 @@
 package com.ajaxproject.warehouse.service
 
-import com.ajaxproject.warehouse.annotation.RateLimit
 import com.ajaxproject.warehouse.dto.ProductCreateDto
 import com.ajaxproject.warehouse.dto.ProductDataDto
 import com.ajaxproject.warehouse.dto.ProductDataLiteDto
 import com.ajaxproject.warehouse.dto.ProductUpdateDto
-import com.ajaxproject.warehouse.entity.Product
+import com.ajaxproject.warehouse.entity.MongoProduct
 import com.ajaxproject.warehouse.exception.NotFoundException
-import com.ajaxproject.warehouse.repository.ProductRepository
+import com.ajaxproject.warehouse.repository.MongoProductRepository
+import org.bson.types.ObjectId
 import org.springframework.stereotype.Service
 
 @Service
-class ProductServiceImpl(val productRepository: ProductRepository) : ProductService {
-    override fun findAllProducts(): List<ProductDataLiteDto> {
-        return productRepository.findAll().map { it.mapToLiteDto() }
-    }
+class ProductServiceImpl(
+    val mongoProductRepository: MongoProductRepository
+) : ProductService {
+
+    override fun findAllProducts(): List<ProductDataLiteDto> =
+        mongoProductRepository.findAll().map { it.mapToLiteDto() }
 
-    @RateLimit
-    override fun findById(id: Int): ProductDataDto {
-        return productRepository.findById(id)
-            .orElseThrow { NotFoundException("Product with id $id not found") }
-            .mapToDataDto()
+    override fun getById(id: String): ProductDataDto {
+        val mongoProduct: MongoProduct = mongoProductRepository.findById(ObjectId(id))
+            ?: throw NotFoundException("Product with id $id not found")
+        return mongoProduct.mapToDataDto()
     }
 
     override fun createProduct(createDto: ProductCreateDto): ProductDataDto {
-        val product = productRepository.save(createDto.mapToEntity())
+        val product: MongoProduct = mongoProductRepository.createProduct(createDto.mapToEntity())
         return product.mapToDataDto()
     }
 
-    override fun updateProduct(updateDto: ProductUpdateDto, id: Int): ProductDataDto {
+    override fun updateProduct(updateDto: ProductUpdateDto, id: String): ProductDataDto {
         require(id == updateDto.id) { "Mapping id is not equal to request body id" }
-        val product: Product = productRepository.findById(id)
-            .orElseThrow { NotFoundException("Product with id $id not found") }
-        product.setUpdatedData(updateDto)
-        productRepository.save(product)
+        var product: MongoProduct = mongoProductRepository.findById(ObjectId(id))
+            ?: throw NotFoundException("Product with id $id not found")
+        product = product.setUpdatedData(updateDto)
+        product = mongoProductRepository.save(product)
```

Comment:
```suggestion
        val product: MongoProduct = mongoProductRepository.findById(ObjectId(id))
            ?: throw NotFoundException("Product with id $id not found")
        val updatedProduct = product.setUpdatedData(updateDto)
        return mongoProductRepository.save(updatedProduct).let { it.mapToDataDto() }
```

Just suggestion.

What I really don't like in your code:
```kotlin
var product = ... 
product = ...
product = ...
```
It's better declaring a separate well-named variable